### PR TITLE
Add HPyErr_SetFromErrno

### PR DESCRIPTION
**Pull request by steve-s**
This is necessary for porting `psutils`

### CODE REVIEW

File: test/test_hpyerr.py
Comment (antocuni) [inline test/test_hpyerr.py:154]: on Python3, `IOError` is just an alias to `OSError`. Why do you need to test both here?

File: test/test_hpyerr.py
Comment (antocuni) [inline test/test_hpyerr.py:158]: I suppose this is needed because `fseek` could return either error, but I had to read its manpage to understand that. Maybe it's worth a short comment to explain

**Review by antocuni — CHANGES_REQUESTED** [review]
LGTM in general but windows tests are failing

File: test/test_hpyerr.py
Comment (steve-s) [reply test/test_hpyerr.py:158]: Yes, good point I'll add some explanatory comment. Maybe I'll have to change this altogether, because it does not work on Windows it seems at least from the azure pipelines.

File: test/test_hpyerr.py
Comment (steve-s) [reply test/test_hpyerr.py:154]: Ah, good point. I wanted to test it with at least two exceptions, so I searched how the API is used in packages and saw two two types being used. I'll find some other suitable type

**Comment by steve-s** [comment]
I forgot that `PyErr_SetFromErrno` also conveniently returns `NULL`. I think it would be nice to have that also in HPy. I'll wait for the CI jobs to finish and then I'll push one more commit with this change:
```
--- a/hpy/tools/autogen/public_api.h
+++ b/hpy/tools/autogen/public_api.h
@@ -179,7 +179,7 @@ HPy HPy_CallTupleDict(HPyContext *ctx, HPy callable, HPy args, HPy kw);
 void HPy_FatalError(HPyContext *ctx, const char *message);
 void HPyErr_SetString(HPyContext *ctx, HPy h_type, const char *message);
 void HPyErr_SetObject(HPyContext *ctx, HPy h_type, HPy h_value);
-void HPyErr_SetFromErrno(HPyContext *ctx, HPy h_type);
+HPy HPyErr_SetFromErrno(HPyContext *ctx, HPy h_type);
```

**Comment by antocuni** [comment]
ok, waiting for your last commit before merging then

**Comment by steve-s** [comment]
> ok, waiting for your last commit before merging then

done (also rebased on master)

**Review by antocuni — APPROVED** [review]

**Pull request merged**
